### DISCUSSION TITLE
Workspace m+1/m-1 behavior

### DISCUSSION THREAD

**Discussion by hvitoi** [hyprwm/Hyprland#2067] [Q&A] [answered]
I'm trying to achieve the following behavior:

- Focus the next workspace (if any) in the same monitor
- If next workspace does not exist, create a new empty workspace in the same monitor and focus on it

This behavior is exactly achieved by using the three finger swipe gesture.
Using keybindings, however, the closer I have got to this is using the dispatcher `workspace m+1`, but with that if the next workspace does not exist, it is swapped with the first workspace.

My questions are:
- Is the current m+1/m-1 behavior the expected one? Modifying its logic in order to achieve the behavior described above would be something desired?
- Do you see any other way around to achieve the mentioned behavior using keybindings before recurring to complex bash scripts?

**Comment by vaxerski** [comment]
> Do you see any other way around to achieve the mentioned behavior using keybindings before recurring to complex bash scripts?

No. Although the script would not really be that complex... Execute m+1 if next workspace exists, +1 otherwise.

**Reply by hvitoi** [reply]
I'm scratching this script and turns out the complexity is due to the multi-monitor setup (get the active monitor, get active workspace, get workspaces array and iterate through its index).

But anyway: adding this behavior in the codebase would be something desirable? By modifying the current m+1 or adding a new foo+1/foo-1 ?

This way of navigating through workspaces is the default on gnome and some people (like me :)) might be looking forward to it.

I believe this function may be already somewhere in the codebase given the three-fingers swipe gesture.

**Reply by vaxerski** [reply]
there may be an open issue about that. I've heard the request somewhere. If not, feel free to open a new one.

**Comment by hvitoi** [comment, marked as answer]
Cross-posting here the current open issue about it https://github.com/hyprwm/Hyprland/issues/1473
There is also reference to a python script to be used as a workaround.
